File: backend/src/main/java/com/workorder/service/WorkOrderService.java
package com.workorder.service;

import com.workorder.dto.WorkOrderRequest;
import com.workorder.model.WorkOrder;

import java.util.List;

public interface WorkOrderService {
    List<WorkOrder> findAll();
    WorkOrder findById(Long id);
    WorkOrder create(WorkOrderRequest request);
    WorkOrder update(Long id, WorkOrderRequest request);
    void delete(Long id);
}

File: frontend/src/app/services/work-order.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { WorkOrder, WorkOrderRequest } from '../models/work-order.model';

@Injectable({ providedIn: 'root' })
export class WorkOrderService {

  private readonly apiUrl = 'http://localhost:8080/api/work-orders';

  constructor(private http: HttpClient) {}

  getAll(): Observable<WorkOrder[]> {
    return this.http.get<WorkOrder[]>(this.apiUrl);
  }

  getById(id: number): Observable<WorkOrder> {
    return this.http.get<WorkOrder>(`${this.apiUrl}/${id}`);
  }

  create(request: WorkOrderRequest): Observable<WorkOrder> {
    return this.http.post<WorkOrder>(this.apiUrl, request);
  }

  update(id: number, request: WorkOrderRequest): Observable<WorkOrder> {
    return this.http.put<WorkOrder>(`${this.apiUrl}/${id}`, request);
  }

  delete(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}

